tools/windows-node-installer/pkg/cloudprovider/interface.go

```diff
@@ -1,14 +1,15 @@
 package cloudprovider
 
-// Cloud interface describes all abstract functions for Cloud interface
-// These functions should be basic and implemented across all supported cloud vendors
-// TODO: CreateWindowsVM should return a vendor session for future interaction with created instance
+// Cloud is the interface that needs to be implemented per provider to allow support for creating Windows nodes on
+// that provider.
 type Cloud interface {
 	// CreateWindowsVM creates windows VM based on available image id, instance type, and key name
-	// Based on these information, the created instance should have specific image and instance type, and access key respectively
-	CreateWindowsVM(imageId, instanceType, keyName *string) error
-	// DestroyWindowsVM consumes windows-node-installer.json file which contains necessary information such as ID and security group of created instance
-	// Based on the information in the file, the DestroyWindowsVM deletes those instances and security group
-	// This function can also serve as clean up operation for destroying instances that failed to completely be created
+	// TODO: CreateWindowsVM should return a vendor session for further interaction with the created instance
+	CreateWindowsVM(imageId, instanceType, keyName string) error
+	// DestroyWindowsVM uses 'windows-node-installer.json' file which contains information such as ID's and
+	// security groups of created instances and deletes them.
+	// It deletes the security groups only if the group is not associated with any instance.
+	// This function can also serve as a clean up operation for destroying instances that failed to be completely
```

Comment:
AWS does not create incomplete instances, so I am not sure what you are implying.